packages/utils/src/date-utils/timeFormatting.ts

```diff
@@ -71,11 +70,80 @@ export function removeSeconds(timer: string): string {
 }
 
 /**
- * @description utility function to format a date in milliseconds using luxon
- * @param {number} millis
- * @param {string} format
- * @return {string}
+ * Formats a given date into a custom string format based on UTC time.
+ * 
+ * @param millis - The number of milliseconds.
+ * @param format - A string specifying the desired output format. 
+ *                 For example, 'yyyy-MM-dd HH:mm:ss' will format the date as '2024-01-23 09:41:08'.
+ * 
+ * @returns The formatted date as a string according to the provided `format` string.
+ * 
+ * @throws Will throw an error if the date formatting fails or the regex match returns null, which should not happen
+ *         if the `Intl.DateTimeFormat` is correctly configured and the input `millis` is valid.
+ * 
+ * @throws Will throw an error if the input @param millis is smaller than zero.
  */
 export function formatFromMillis(millis: number, format: string): string {
-  return DateTime.fromMillis(millis).toUTC().toFormat(format);
+  if (millis < 0) {
+    throw new Error("Input `millis` can't be smaller than zero.");
+  }
+
+  const date: Date = new Date(millis);
+  const options: Intl.DateTimeFormatOptions = {
+      year: 'numeric',
+      month: '2-digit',
+      day: '2-digit',
+      hour: '2-digit',
+      minute: '2-digit',
+      second: '2-digit',
+      hour12: false,
+      hourCycle: 'h23', // Ensures the hour is always two digits
+      timeZone: 'UTC'
+  };
+
+  const formattedDate: string = date.toLocaleString('en-US', options);
+  const milliseconds: string = date.getUTCMilliseconds().toString().padStart(3, '0');
+
+  // Extract date and time components.
+  const match = formattedDate.match(/(\d{2})\/(\d{2})\/(\d{4}), (\d{2}):(\d{2}):(\d{2})/);
+  if (!match) {
+      throw new Error("Date format mismatch or null result from regex match.");
+  }
+
+  const [_, month, day, year, hour24, minute, second] = match;
```

Comment:
Nice!
This logic is well structured and pretty clever.

I wonder if the maintenance would be made easier by avoiding the regex and instead having explicit logic for the datasets. something like (untested)


```ts
  const date: Date = new Date(millis);
  const year = date.getUTCFullYear();
  const month = (date.getUTCMonth() + 1).toString().padStart(2, '0');
  const day = date.getUTCDate().toString().padStart(2, '0');
  const hour24 = date.getUTCHours().toString().padStart(2, '0');
  const minute = date.getUTCMinutes().toString().padStart(2, '0');
  const second = date.getUTCSeconds().toString().padStart(2, '0');
  const milliseconds = date.getUTCMilliseconds().toString().padStart(3, '0');

  const hour12 = (date.getUTCHours() % 12 || 12).toString();
```

We can probably remove `year` `month` and `day` since there is no expectation of us having to deal with it (app has no days)

Reply:
I refactored the regex to your recommendation and removed `year` `month` and `day`